addons/mail/controllers/thread.py

```diff
@@ -8,6 +8,22 @@
 
 
 class ThreadController(http.Controller):
+    @http.route("/mail/thread/access_rights", methods=["POST"], type="json", auth="user")
+    def mail_thread_access_rights(self, thread_data):
+        res = []
+        for thread_model in thread_data:
+            threads = request.env[thread_model].with_context(active_test=False).search([("id", "in", thread_data[thread_model])])
+            access_rights = threads._get_mail_thread_access_rights()
+            for thread_id in thread_data[thread_model]:
+                res.append(
+                    {
+                        "id": thread_id,
+                        "model": thread_model,
+                        **access_rights.get((thread_model, thread_id), {"hasReadAccess": False, "hasWriteAccess": False}),
+                    }
+                )
```

Comment:
With other suggestions, could become
```suggestion
            res += [
                dict(id=thread.id, model=thread_model, **ar_data)
                for thread, ar_data in zip(threads, threads._get_mail_thread_data_access_rights())
            ]
```